### PR TITLE
Generate CLI docs on make command

### PR DESCRIPTION
**Pull request by mukundansundar**
Add gen docs and diff checking to workflow

# Description

Generate docs automatically on make command. 

Error out if the docs are different or new files are added in workflow. Similar to go mod diff.  

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -54,6 +54,12 @@ Run sidecar only:
 		viper.BindPFlag("placement-host", cmd.Flags().Lookup("placement-host"))
 	},
 	Run: func(cmd *cobra.Command, args []string) {
+		if configFile == "" {
```
Comment (mukundansundar) [inline cmd/run.go:57]: This is checked here and not added in defaults so that the automatic docs generated do not have any platform dependent defaults.

Comment (mukundansundar) [reply cmd/run.go:57]: https://github.com/dapr/cli/runs/1009642534?check_suite_focus=true#step:8:28

File: .github/scripts/docs.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"github.com/dapr/cli/cmd"
+	"github.com/spf13/cobra/doc"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		panic("Requires a path to generate docs in.")
+	}
+	// Tag contains date. Need to disable generation to avoid date generation errors.
+	cmd.RootCmd.DisableAutoGenTag = true
```
Comment (mukundansundar) [inline .github/scripts/docs.go:20]: Example of error: https://github.com/dapr/cli/pull/446/checks?check_run_id=1010866665#step:8:89

File: docs/development/development.md [hunk 3]
```diff
@@ -70,6 +70,12 @@ make build GOOS=linux GOARCH=amd64
 make test
 ```
 
+### Generate CLI docs
+
+```bash
```
Comment (amanbha) [inline docs/development/development.md:75]: Can we add the steps here on what is needed to checkin the docs when new command is added or existing one is changed?

**Pull request closed without merging**